README.md

```diff
@@ -643,3 +643,60 @@ macro:
   elapsed_sec:
     mean: 25.911653497483996
 ```
+
+## Retrieval Evaluation Metrics
+
+### Recall@k
+
+**Recall@k** tells you what fraction of the total number of relevant items were found in the top 'k' recommendations. It answers the question: "Out of all the items the user actually cares about, how many did we successfully show them in the first k spots?"
+
+* **How it works**: You count the number of relevant items in the top `k` retrieved results and divide that by the *total* number of items that are actually relevant.
+* **Formula**:
+    $$
+    \text{Recall@k} = \frac{\text{Number of relevant items in top k}}{\text{Total number of relevant items}}
+    $$
+* **Example**: If there are **5** total relevant documents for a query and your system retrieves **3** of them in the top 10 results (`k=10`), your Recall@10 is `3 / 5 = 0.6`.
+
+```python
+from retrieval_evaluation import recall_at_k
+relevant_items = {1, 3, 5, 7, 9}
+retrieved_items = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+
+k_value = 5
+r_at_k = recall_at_k(relevant_items, retrieved_items, k_value)
+
+print(f"Recall@{k_value}: {r_at_k}") # Recall@5: 0.6
+```
+
+### Average Precision (AP)
+
+**Average Precision (AP)** evaluates a ranked list of recommendations by looking at the precision at the position of each correctly retrieved item. It rewards systems for placing relevant items higher up in the list. It's more sophisticated than just looking at precision at a single cutoff because it considers the entire ranking.
+
+* **How it works**: For a single query, you go down the list of retrieved items. Every time you encounter a relevant item, you calculate the precision *at that exact spot* (i.e., `number of hits / current rank`). You then average all of these precision scores. The final average is divided by the total number of relevant items.
+* **Formula**:
+    $$
+    \text{AP} = \frac{1}{\text{Total number of relevant items}} \sum_{k=1}^{n} (P(k) \times \text{rel}(k))
```

Comment:
Not rendered properly 
![image](https://github.com/user-attachments/assets/df3c0479-6d50-43da-ba77-609b83b5abb4)

Reply:
Fixed